plutus-ledger-api/src/Plutus/V1/Ledger/Contexts.hs

```diff
@@ -88,14 +90,22 @@ redeemer and data scripts of all of its inputs and outputs.
 data TxInInfo = TxInInfo
     { txInInfoOutRef   :: TxOutRef
     , txInInfoResolved :: TxOut
-    } deriving (Generic)
+    } deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty TxInInfo where
+    pretty TxInInfo{txInInfoOutRef, txInInfoResolved} =
+        pretty txInInfoOutRef <+> "->" <+> pretty txInInfoResolved
 
 -- | Purpose of the script that is currently running
 data ScriptPurpose
     = Minting CurrencySymbol
     | Spending TxOutRef
     | Rewarding StakingCredential
     | Certifying DCert
+    deriving stock (Generic, Haskell.Show, Haskell.Eq)
+
+instance Pretty ScriptPurpose where
```

Comment:
we have a `PrettyShow` newtype for this so you can do `deriving Pretty via (PrettyShow ScriptPurpose)`. Not a big deal though.